src/qibo/hamiltonians/abstract.py

```diff
@@ -65,37 +76,45 @@ def exp(self, a):  # pragma: no cover
         raise_error(NotImplementedError)
 
     @abstractmethod
-    def expectation(self, state, normalize=False):  # pragma: no cover
-        """Computes the real expectation value for a given state.
+    def expectation(self, circuit, nshots: Optional[int] = None):  # pragma: no cover
+        """Computes the expectation value for a given circuit.
 
         Args:
-            state (ndarray): state in which to calculate the expectation value.
-            normalize (bool, optional): If ``True``, the expectation value
-                :math:`\\ell_{2}`-normalized. Defaults to ``False``.
+            circuit (Circuit): circuit to calculate the expectation value from.
+        If the circuit has already been executed, this will just make use of the cached
+        result, otherwise it will execute the circuit.
```

Comment:
```suggestion
                If the circuit has already been executed, this will just make use of the cached
                result, otherwise it will execute the circuit.
```